test/subtype.jl

```diff
@@ -1229,3 +1235,22 @@ struct A23764{T, N, S} <: AbstractArray{Union{T, S}, N}; end
 struct A23764_2{T, N, S} <: AbstractArray{Union{Ref{T}, S}, N}; end
 @test Tuple{A23764_2{T, 1, Void} where T} <: Tuple{AbstractArray{T,N}} where {T,N}
 @test Tuple{A23764_2{T, 1, Void} where T} <: Tuple{AbstractArray{T,N} where {T,N}}
+
+# issue #24305
+f24305(x) = [g24305(x) g24305(x) g24305(x) g24305(x); g24305(x) g24305(x) 0 0];
+@test_throws UndefVarError f24305(1)
+
+f1_24305(x,y,z) = x*y-z^2-1
+f2_24305(x,y,z) = x*y*z+y^2-x^2-2
+f3_24305(x,y,z) = exp(x)+z-exp(y)-3
+Fun_24305(x) = [ f1_24305(x[1],x[2],x[3]); f2_24305(x[1],x[2],x[3]); f3_24305(x[1],x[2],x[3]) ]
```

Comment:
adding spaces to the above expressions to make them readable would be appreciated